### ISSUE TITLE
gcr-cleaner deletes the most recent images instead of oldest

### ISSUE THREAD

**Issue by ahsan-raza** [GoogleCloudPlatform/gcr-cleaner#138]
### TL;DR

If I run the gcr-cleaner with the following arguments, it deletes the most recent images and keep the oldest image which is contradictory to what its described in the docs

`./gcr-cleaner-cli -repo europe-west2-docker.pkg.dev/<my-group>/devops/root  -dry-run  -recursive  -keep 1  -tag-filter-any "^.*$" `

### Expected behavior

Should delete the oldest image first and keep the most recent images.

### Observed behavior

It deletes the recent images instead of oldest image.

### Debug log output

```text
Deleting refs older than 2023-03-20T10:24:27Z on 2 repo(s)...

europe-west2-docker.pkg.dev/<my-group>/devops/root
  ✓ 123
  ✓ 234
  ✓ 345
  ✓ develop
  ✓ latest
  ✓ sha256:34a1fd2c9e47f4262894d24221f188542eb6acd19629bcf6084af4fb3550a775
  ✓ sha256:3afb7c1361e66b8225bd447b1838f9d4f71d3114f7184488c5dad1857d836100
  ✓ sha256:767ef147da916a872ede3ea699a076a20b5bcd6fe4a036e7fcf90507b847df09
  ✓ sha256:7de4a165eb7bbd98247212be7c7151b42e4565bb4e080a78160648941cefe8c0
  ✓ sha256:a78aad418907c36327ef98acebff884aabe258f2b5067c9c79f754c3528a7f72
  ✓ v1.0.0-20
```

### Additional information

while the timestamp of the images present in the registry is below:

<img width="606" alt="image" src="https://user-images.githubusercontent.com/11913516/226314394-c02bfa88-a771-4462-a432-00d82b65b6af.png">

**Comment by sethvargo** [comment]
Filtering out duplicate data, I think gcr-cleaner is behaving as expected here. Look at the `created` timestamps in the API response. The entries are sorted by created DESC and then we keep the most recent 3 - the most recent 3 are "latest", "v1.0.0-20", and "v1.0.0-21":

```json
[
  {
    "digest": "sha256:9547c6",
    "message": "processing manifest",
    "tags": [
      "latest"
    ],
    "created": "2023-03-20T11:58:03+01:00",
    "uploaded": "2023-03-20T11:58:41+01:00"
  },
  {
    "digest": "sha256:9547c6",
    "message": "should delete",
    "reason": "matches tag filter",
    "tags": [
      "latest"
    ]
  },
  {
    "digest": "sha256:9547c6",
    "keep": 3,
    "keep_count": 0,
    "message": "skipping deletion because of keep count",
    "created": "2023-03-20T11:58:03+01:00",
    "uploaded": "2023-03-20T11:58:41+01:00"
  },
  {
    "digest": "sha256:34a1fd",
    "message": "processing manifest",
    "tags": [
      "v1.0.0-20"
    ],
    "created": "2023-03-17T11:59:22+01:00",
    "uploaded": "2023-03-20T11:22:49+01:00"
  },
  {
    "digest": "sha256:34a1fd",
    "message": "should delete",
    "reason": "matches tag filter",
    "tags": [
      "v1.0.0-20"
    ]
  },
  {
    "digest": "sha256:34a1fd",
    "keep": 3,
    "keep_count": 1,
    "message": "skipping deletion because of keep count",
    "created": "2023-03-17T11:59:22+01:00",
    "uploaded": "2023-03-20T11:22:49+01:00"
  },
  {
    "digest": "sha256:767ef1",
    "message": "processing manifest",
    "tags": [
      "v1.0.0-21"
    ],
    "created": "2023-03-17T11:59:22+01:00",
    "uploaded": "2023-03-20T09:10:33+01:00"
  },
  {
    "digest": "sha256:767ef1",
    "message": "should delete",
    "reason": "matches tag filter",
    "tags": [
      "v1.0.0-21"
    ]
  },
  {
    "digest": "sha256:767ef1",
    "keep": 3,
    "keep_count": 2,
    "message": "skipping deletion because of keep count",
    "created": "2023-03-17T11:59:22+01:00",
    "uploaded": "2023-03-20T09:10:33+01:00"
  },
  {
    "digest": "sha256:3afb7c",
    "message": "processing manifest",
    "tags": [
      "234"
    ],
    "created": "2023-03-17T11:54:49+01:00",
    "uploaded": "2023-03-20T11:22:54+01:00"
  },
  {
    "digest": "sha256:3afb7c",
    "message": "should delete",
    "reason": "matches tag filter",
    "tags": [
      "234"
    ]
  },
  {
    "digest": "sha256:7de4a1",
    "message": "processing manifest",
    "tags": [
      "345"
    ],
    "created": "2023-03-03T14:53:25+01:00",
    "uploaded": "2023-03-20T11:22:52+01:00"
  },
  {
    "digest": "sha256:7de4a1",
    "message": "should delete",
    "reason": "matches tag filter",
    "tags": [
      "345"
    ]
  },
  {
    "digest": "sha256:a78aad",
    "message": "processing manifest",
    "tags": [
      "123",
      "develop"
    ],
    "created": "2022-09-26T21:56:43+02:00",
    "uploaded": "2023-03-20T11:22:54+01:00"
  },
  {
    "digest": "sha256:a78aad",
    "message": "should delete",
    "reason": "matches tag filter",
    "tags": [
      "123",
      "develop"
    ]
  },
  {
    "digest": "sha256:3afb7c",
    "message": "deleting tag",
    "tag": "234"
  },
  {
    "digest": "sha256:a78aad",
    "message": "deleting tag",
    "tag": "123"
  },
  {
    "digest": "sha256:7de4a1",
    "message": "deleting tag",
    "tag": "345"
  },
  {
    "digest": "sha256:a78aad",
    "message": "deleting tag",
    "tag": "develop"
  },
  {
    "digest": "sha256:3afb7c",
    "message": "deleting digest",
    "severity": "DEBUG"
  },
  {
    "digest": "sha256:a78aad",
    "message": "deleting digest",
    "severity": "DEBUG"
  },
  {
    "digest": "sha256:7de4a1",
    "message": "deleting digest",
    "severity": "DEBUG"
  }
]
```


This matches the documented behavior.
